satellite/metainfo/metainfo.go

```diff
@@ -1149,7 +1149,7 @@ func (endpoint *Endpoint) CommitObject(ctx context.Context, req *pb.ObjectCommit
 	lastSegmentPointer.Remote.Redundancy = streamID.Redundancy
 	lastSegmentPointer.Metadata = req.EncryptedMetadata
 
-	err = endpoint.metainfo.Delete(ctx, lastSegmentPath)
+	err = endpoint.metainfo.UnsynchronizedDelete(ctx, lastSegmentPath)
```

Comment:
I just realized that we need to use here `Delete`.

Reply:
I think the endpoints are fine since we don't yet support concurrent uploads and deletes to the same path, but I can do the change nevertheless.

Reply:
change is easy so make sense to do this sooner than later